### ISSUE TITLE
avscms installation tutorial

### ISSUE THREAD

**Issue by tianchengdemo** [avscms/avscms#12]
Excuse me, do you happen to have any installation tutorials available? Additionally, could you provide me with the pseudo-static rules for nginx?

**Comment by cronner** [comment]
No install guide, I use PHP 7.4 with mariadb, manually imported DB

Nginx rewrite:

        location / {
        index index.html index.php;
        try_files $uri $uri/ /loader.php?$args;
        }

**Comment by tianchengdemo** [comment]
> mariadb

Thank you for sharing this information with me. However, may I please inquire about the default administrator account username and password for the imported database?

**Comment by cronner** [comment]
and btw do something in the nginx site config for streaming the mp4 files if you host files:

location ~* .mp4$ {

    # Enable Range requests for seeking within the video
    # This is important for streaming and video players
    mp4;
    mp4_buffer_size 1m;
    mp4_max_buffer_size 10m;
    sendfile on;
    sendfile_max_chunk 512k;
    aio threads;
  }

**Comment by tianchengdemo** [comment]
> and btw do something in the nginx site config for streaming the mp4 files if you host files:
> 
> location ~* .mp4$ {
> 
> ```
> # Enable Range requests for seeking within the video
> # This is important for streaming and video players
> mp4;
> mp4_buffer_size 1m;
> mp4_max_buffer_size 10m;
> sendfile on;
> sendfile_max_chunk 512k;
> aio threads;
> ```
> 
> }
Thank you

**Comment by klkuno** [comment]
Hi everyone. I have to add FFMpeg, in my server, but my website can't find it. Patch for FFMpeg is /usr/local/bin/ffmpeg, FFProbe: /usr/local/bin/ffprobe. Is correct? Im use openlitespeed, I have the same problem with PhpName, missing. /usr/local/bin/php. Open Basedir report a problem. Thanks to anyone who can help me

**Comment by dphilp1** [comment]
I'm a complete noob here. I've tried setting up a server but I get lost when it says things like: 

Apache Web Server

Support for AcceptPathInfo
Support for MultiViews (or mod_rewrite which is slower)

or same for php. I followed along a DigitalOCean walkthrough for a LAMP stack, got mysql installed and working, uploaded the sql database with phpmyadmin. Then it says add all files and folders from /uploads.....where is /uploads? I don't have that in my avscms folder anywhere but in modules....So I uploaded everything to the root folder of the domain......when I tried to access my_domain/siteadmin I was getting 403 errors, so I changed file permissions, now I'm getting "This page isn't working".....I wish there was a noob friendly walkthrough.....I've messed something up but I don't know where...

**Comment by klkuno** [comment]
> No install guide, I use PHP 7.4 with mariadb, manually imported DB
> 
> Nginx rewrite:
> 
> ```
>     location / {
>     index index.html index.php;
>     try_files $uri $uri/ /loader.php?$args;
>     }
> ```

Could you show me your php.ini?
I currently have: php 7.4 gd, fastcgi, Curl, safe_mode off, open_basedir off, max_execution_time 7200, max_input_time 7200, exec disable, ffmpeg all ok. The whole site works, but the videos remain in the queue. I think it's something to configure in php 7

**Comment by cronner** [comment]
AVS specific settings:

php_admin_value[memory_limit]  = 1025M
php_admin_value[max_execution_time] = 7200
php_admin_value[upload_max_filesize] = 500M
php_admin_value[max_input_time] = 7200
php_admin_value[post_max_size] = 500M
php_flag[safe_mode] = off
php_admin_value[open_basedir] = "/home/site.com:/usr/share/pear:/usr/share/php:/tmp:/usr/local/lib/php:/usr/bin/php7.4:/usr/bin/ffmpeg:/usr/bin/ffprobe"

**Comment by klkuno** [comment]
I added those parameters, but the videos still remain in the queue. Yet ffmpeg is seen by the system, I don't understand why the video isn't processed

**Comment by cronner** [comment]
there is some problems sometimes with queue but mostly it works, maybe permissions?

**Comment by klkuno** [comment]
Which permissions are you referring to? I can check

**Comment by cronner** [comment]
all folder permission they are also green or?

**Comment by klkuno** [comment]
Yes, is all green 
<img width="1392" alt="perm" src="https://github.com/avscms/avscms/assets/96708235/2fcdaebf-5eac-42ed-aa17-927520710f67">

**Comment by klkuno** [comment]
I solved the problem, but I noticed that another problem. if you upload more than two videos at a time the videos remain in the queue. Have set Max Simultaneous Conversions: 2, or max 100, 2 ok work, from three onwards the videos remain in place forever. This script was also paid for some time ago, but it is full of problems. I advise against its use. There is much better out there

**Comment by cronner** [comment]
maybe type the sollution for others to use? :)

**Comment by klkuno** [comment]
Don't use Ubuntu 22, but 20. But then there are many other problems to solve, I do not recommend its use

**Comment by cronner** [comment]
I use Debian with multimedia repo

**Comment by klkuno** [comment]
Have you tried uploading 3 files of around 70 MB at the same time?

**Comment by cronner** [comment]
yeah seems to work but sometimes it just fails :/

**Comment by klkuno** [comment]
Exactly, considering that 3 files are too few, imagine 50 files at the same time 😄

**Comment by superpitzi2** [comment]
> I solved the problem, but I noticed that another problem. if you upload more than two videos at a time the videos remain in the queue. Have set Max Simultaneous Conversions: 2, or max 100, 2 ok work, from three onwards the videos remain in place forever. This script was also paid for some time ago, but it is full of problems. I advise against its use. There is much better out there

How did you solved, I have the same problem, I am with ubuntu 22 and I did everything but its not going through on video conversion, keep staying there. it does not get the thumbnails, anyone know this error ? thanks

**Comment by fernnje** [comment]
> No install guide, I use PHP 7.4 with mariadb, manually imported DB
> 
> Nginx rewrite:
> 
> ```
>     location / {
>     index index.html index.php;
>     try_files $uri $uri/ /loader.php?$args;
>     }
> ```

How I can install it in litespeed server? I can't access all page without htaccess

**Comment by royalphp** [comment]
@fernnje maybe you should try a pre-built project like this one here: https://github.com/royalphp/avscms-docker

**Comment by fernnje** [comment]
> [@fernnje](https://github.com/fernnje) maybe you should try a pre-built project like this one here: https://github.com/royalphp/avscms-docker

I was already to fix it, but now I have problem with upload folder because I have delete it, you know to generate upload folder?

**Comment by royalphp** [comment]
> > [@fernnje](https://github.com/fernnje) maybe you should try a pre-built project like this one here: https://github.com/royalphp/avscms-docker
> 
> I was already to fix it, but now I have problem with upload folder because I have delete it, you know to generate upload folder?

I don't think the problem is with the deleted folder. Probably the problem is that you are using a web server that is not supported by this project.
This project is too old for new technologies and versions. The best way to deploy this is with Apache 2.4 web server and PHP 5.6. These are the technologies and versions on which this project began to be developed, and have the best compatibility.
